doc/source/reference/arrays.rst

```diff
@@ -45,5 +45,6 @@ of also more complicated arrangements of data.
    arrays.nditer
    arrays.classes
    maskedarray
+   recordarray
```

Comment:
I'd vote to remove the mention from the top-level array refguide article, just because I don't think recarrays are recommended for new code.

Reply:
It can be removed. But a "document isn't included in any toctree" warning will be raised. I don't think adding it into routines.rst is a better choice.